docs/changelog/v6.md

```diff
@@ -0,0 +1,214 @@
+
+```purs
+HH.div_
+  [ HH.button
+      [ HE.onClick \_ -> Just Clear ]
+      [ HH.text "Clear" ]
+  , HH.slot _id unit component unit (Just <<< Handle)
+  ]
+```
+
+becomes this code in Halogen 6:
+
+```purs
+HH.div_
+  [ HH.button
+      [ HE.onClick \_ -> Clear ]
+      [ HH.text "Clear" ]
+  , HH.slot _id unit component unit Handle
+  ]
+```
+
+You're no longer able to ignore the output of a child component by providing a handler `\_ -> Nothing`. Instead, you can use the `slot_` function if you don't care about a child component's output. This code from Halogen 5:
+
+```purs
+HH.slot _id unit component unit (\_ -> Nothing)
+```
+
+becomes this code in Halogen 6:
+
+```purs
+HH.slot_ _id unit component unit
+```
+
+Added in [#636](https://github.com/purescript-halogen/purescript-halogen/pull/636/) and [#642](https://github.com/purescript-halogen/purescript-halogen/pull/642).
+
+## Query Helper Functions
+
+We've simplified the helper functions which are used with queries so that you can use `tell` and `request` directly, rather than use them in conjunction with the `query` and `request` functions.
+
+In Halogen 5, to execute a query you would use the `query` function and combine it with the `request` function (for request-style queries, which return a result) or the `tell` function (for tell-style queries, which don't return a result). This was always a bit difficult to explain and easy to trip over when writing queries yourself.
+
+Here's how you would execute a request-style and then a tell-style query in Halogen 5:
+
+```purs
+handleAction = do
+  a <- H.query _a unit (H.request Child.SomeRequestQuery)
+  _ <- H.query _a unit (H.tell Child.SomeTellQuery)
+```
+
+In Halogen 6, you no longer use the `query` function. Instead, you use `request` and `tell` directly. You also don't have to throw away the result of `tell`, as it can already be safely discarded:
+
+```purs
+handleAction = do
+  a <- H.request _a unit Child.SomeRequestQuery
+  H.tell _a unit Child.SomeTellQuery
+```
+
+The old `tell` and `request` functions still exist in Halogen 6, but they've been renamed to `mkTell` and `mkRequest` and are only used when querying the root of your application. For example, this code in Halogen 5:
+
+```purs
+io <- runUI component unit body
+
+state <- io.query $ H.request SomeRequestQuery
+_ <- io.query $ H.tell SomeTellQuery
+```
+
+becomes this code in Halogen 6:
+
+```purs
+io <- runUI component unit body
+
+state <- io.query $ H.mkRequest SomeRequestQuery
+_ <- io.query $ H.mkTell SomeTellQuery
+```
+
+Added in [#621](https://github.com/purescript-halogen/purescript-halogen/pull/621).
+
+## Event Sources
+
+Event sources have been implemented with `Event` from the `event` library instead of the Halogen event sources implementation on top of `coroutines`. This simplifies the library internals and connects Halogen with a more widely-used FRP library.
```

Comment:
We might want to add a link to that repo's issue tracker, so that people can become more aware of the current issues/bugs in this new dependency. It might save them some time.